distribution/pull_v2.go

```diff
@@ -147,6 +149,10 @@ func (ld *v2LayerDescriptor) DiffID() (layer.DiffID, error) {
 	return ld.V2MetadataService.GetDiffID(ld.digest)
 }
 
+func (ld *v2LayerDescriptor) ForeignSource() *distribution.Descriptor {
+	return ld.foreignSrc
```

Comment:
We need to remember its descriptor so that we can construct a manifest for this image later. `ForeignSource()` is used to retrieve that descriptor when registering the layer.